### PR TITLE
new Tables, Join Tables and Queries

### PR DESCRIPTION
**Pull request by gersonhiggins**
New Queries:
image
image

Merging Feature Join Tables into develop
-Added tables Vets, Specializations and visits:
-Added insert into New tables
-Added new Queries

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -8,3 +8,23 @@ CREATE TABLE animals (
...
+    id                  SERIAL PRIMARY KEY,
+    name                TEXT,
+    age                 INTEGER,
+    date_of_graduation  DATE
+);
+CREATE TABLE specializations (
+    vet_id     INTEGER REFERENCES vets(id),
+    species_id INTEGER REFERENCES species(id),
+    PRIMARY KEY (vet_id, species_id)
+);
```
Comment (Zuhaib042) [inline schema.sql]: - [ ] Just as you created this join table `specializations`, kindly consider creating another join table called `visits` that will complete the many-to-many relationship between vets and animals. `visits` will have vets(id) and animals(id) as references and also `visit_date date` for date of visit.
requirement: 
  - There is a many-to-many relationship between the tables animals and vets: an animal can visit multiple vets and one vet can be visited by multiple animals. Create a "join table" called visits to handle this relationship, it should also keep track of the date of the visit.

File: queries.sql [hunk 2]
```diff
@@ -133,4 +133,68 @@ GROUP BY owners.full_name
...
+JOIN visits ON animals.id = visits.animal_id
+JOIN vets ON visits.vet_id = vets.id
+WHERE vets.name = 'Vet William Tatcher'
+ORDER BY visits.visit_date DESC
+LIMIT 1;
+
+SELECT COUNT(DISTINCT visits.animal_id)
+FROM visits
+JOIN vets ON visits.vet_id = vets.id
+WHERE vets.name = 'Vet Stephanie Mendez';
```
Comment (Zuhaib042) [inline queries.sql:148]: - [ ] Stephanie Mendez visited 4 animals but your query resulted in `3`, kindly consider fixing it.

File: queries.sql [hunk 3]
```diff
@@ -133,4 +133,68 @@ GROUP BY owners.full_name
...
+FROM vets
+LEFT JOIN specializations ON vets.id = specializations.vet_id
+LEFT JOIN species ON specializations.species_id = species.id;
+
+SELECT animals.name
+FROM animals
+JOIN visits ON animals.id = visits.animal_id
+JOIN vets ON visits.vet_id = vets.id
+WHERE vets.name = 'Vet Stephanie Mendez'
+      AND visits.visit_date BETWEEN '2020-04-01' AND '2020-08-30';
```
Comment (Zuhaib042) [inline queries.sql:160]: - [ ] Stephanie Mendez visited `Agumon and Blossom` between these two dates but your query resulted in `0 rows returned` so kindly consider fixing it.

**Review by Zuhaib042 — CHANGES_REQUESTED** [review]
Hi @gersonhiggins 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✅ Well written queries
✅ Screenshots added of all queries

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gersonhiggins** [comment]
image
image
changes made from past reviewer

**Review by zainsadaqat — APPROVED** [review]
# Pull Request Approved ✅ 🎊 🎉 
## Hello @gersonhiggins  👋 

### Things to appreciate
- [ ] Implemented the feature requested by the previous code reviewer 👍 

### Your project is complete!  There is nothing else to say other than it's time to merge it.

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 ☑️ 



Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**